pkg/server/http/svc_test.go

```diff
@@ -0,0 +1,1123 @@
+	if err != nil {
+		log.Fatal("could not register the Groups handler")
+	}
+
+	err = service.Server().Start()
+	if err != nil {
+		log.Fatalf("could not start server: %v", err)
+	}
+}
+
+func cleanUp(t *testing.T) {
+	datastore := filepath.Join(dataPath, "accounts")
+
+	files, err := ioutil.ReadDir(datastore)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, f := range files {
+		found := false
+		for _, defUser := range DefaultUsers {
+			if f.Name() == defUser {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			deleteAccount(t, f.Name())
+		}
+	}
+}
+
+func sendRequest(method, endpoint, body, auth string) (*httptest.ResponseRecorder, error) {
+	var reader = strings.NewReader(body)
+	req, err := http.NewRequest(method, endpoint, reader)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	if auth != "" {
+		req.Header.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(auth)))
+	}
+
+	rr := httptest.NewRecorder()
+
+	service := getService()
+	service.ServeHTTP(rr, req)
+
+	return rr, nil
+}
+
+func getService() svc.Service {
+	c := &config.Config{
+		HTTP: config.HTTP{
+			Root:      "/",
+			Addr:      "localhost:9110",
+			Namespace: "com.owncloud.web",
+		},
+		TokenManager: config.TokenManager{
+			JWTSecret: "HELLO-secret",
+		},
+		Log: config.Log{
+			Level: "debug",
+		},
+	}
+
+	var logger ocisLog.Logger
+
+	svc := svc.NewService(
+		svc.Logger(logger),
+		svc.Config(c),
+	)
+
+	return svc
+}
+
+func createUser(u User) error {
+	_, err := sendRequest(
+		"POST",
+		"/v1.php/cloud/users?format=json",
+		u.getUserRequestString(),
+		"admin:admin",
+	)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func TestCreateUser(t *testing.T) {
+	testData := []struct {
+		user User
+		err  *Meta
+	}{
+		// A simple user
+		{
+			User{
+				Enabled:     "true",
+				Username:    "rutherford",
+				ID:          "rutherford",
+				Email:       "rutherford@example.com",
+				Displayname: "ErnestRutherFord",
+			},
+			nil,
+		},
+
+		// User with Uid and Gid defined
+		{
+			User{
+				Enabled:     "true",
+				Username:    "thomson",
+				ID:          "thomson",
+				Email:       "thomson@example.com",
+				Displayname: "J. J. Thomson",
+				UIDNumber:   20027,
+				GIDNumber:   30000,
+			},
+			nil,
+		},
+
+		// User with different username and Id
+		{
+			User{
+				Enabled:     "true",
+				Username:    "niels",
+				ID:          "bohr",
+				Email:       "bohr@example.com",
+				Displayname: "Niels Bohr",
+			},
+			nil,
+		},
+
+		// User with special character in username
+		{
+			User{
+				Enabled:     "true",
+				Username:    "schrödinger",
+				ID:          "schrödinger",
+				Email:       "schrödinger@example.com",
+				Displayname: "Erwin Schrödinger",
+			},
+			&Meta{
+				Status:     "error",
+				StatusCode: 400,
+				Message:    "preferred_name 'schrödinger' must be at least the local part of an email",
```

Comment:
that error sounds strange and seem not have to do anything with the real issue, is that a bug?

Reply:
I thought that was because of the special character in the username. I am not totally sure what the criteria for a valid username is.

Reply:
that error sounds to me that `schrödinger@example.com` does not contain `schrödinger` but it does, maybe open an issue and link it here